scio-extra/src/test/scala/com/spotify/scio/extra/sparkey/SparkeyTest.scala

```diff
@@ -68,11 +102,19 @@ class SparkeyTest extends PipelineSpec {
   val bigSideData: IndexedSeq[(String, String)] =
     (0 until 100).map(i => (('a' + i).toString, i.toString))
 
+  "JobTest" should "support mocking sparkey" in {
+    val input = Map("a" -> "b", "c" -> "d")
+    JobTest[SparkeyJob.type]
+      .args("--input=foo", "--output1=bar", "--output2=baz")
+      .input(SparkeyIO("foo"), Seq(MockStringSparkeyReader(input)))
```

Comment:
Can we instead have `SparkeyIO` expect a `Map[K, V]` (or `Seq[(K, V)]`) instead of a reader ? We would create the reader in the IO after `isTest` is checked.

Reply:
I tried this initially; the type parameter on `ScioIO` has to match the type of the mock data iterable passed to `input`, then in the internals of `JobTest` the SCollection returned must also be of the same type, so without changing the contract of ScioIO I don't think this is possible